src/models/User.ts

```diff
@@ -0,0 +1,23 @@
+import mongoose, { Document } from 'mongoose';
+
+export interface User extends Document {
+  email: string;
+  firstName?: string;
+  lastName?: string;
+  password: string;
+  role: 'user' | 'creator';
+  posts?: mongoose.Types.ObjectId[];

```

Comment:
Given a user can have a lot of posts I think we want Posts to be its own table and then reference from it to here. Or can Mongo handle this natively